### PR TITLE
Fix handling of streams with no audio tracks

### PR DESCRIPTION
**Pull request by SimonBrandner**

### CODE REVIEW

File: src/video-grid/useMediaStream.ts [hunk 1]
```diff
@@ -213,7 +213,7 @@ export const useSpatialMediaStream = (
   const sourceRef = useRef<MediaStreamAudioSourceNode>();
 
   useEffect(() => {
-    if (spatialAudio && tileRef.current && !mute) {
+    if (spatialAudio && audioContext && tileRef.current && !mute) {
```
Comment (robintown) [inline src/video-grid/useMediaStream.ts:216]: audioContext is intended to always be present as per the type, so I don't think this is necessary?

**Review by robintown — COMMENTED** [review]
I think it might be nicer to make useSpatialMediaStream work with 0 audio tracks on its own, so the hook doesn't have to be re-run everytime someone unmutes (by reverting my previous change)

File: src/video-grid/useMediaStream.ts (hunk 1 above)
Comment (SimonBrandner) [reply src/video-grid/useMediaStream.ts]: I've seen an issue where it is not :/ Might be related to my SFU branch - will try to track it down and send a fix if it is relevant for regular EC

**Comment by SimonBrandner** [comment]
> I think it might be nicer to make useSpatialMediaStream work with 0 audio tracks on its own, so the hook doesn't have to be re-run everytime someone unmutes (by reverting my previous change)

Good point!

**Review by robintown — APPROVED** [review]
Good idea separating that out into a reusable hook

**Pull request merged**